Makefile

```diff
@@ -10,9 +8,91 @@ RM=--rm
 RUN_FLAGS=-ti
 COVERAGE=
 
-DOCKERHUB_USER=weaveworks
-WEAVE_VERSION=git-$(shell git rev-parse --short=12 HEAD)
+# This specifies the architecture we're building for
+ARCH?=amd64
+
+# We're using QEMU to be able to build Docker images for other architectures
+# from amd64 machines.
+QEMU_VERSION=v2.7.0
+
+# A list of all supported architectures here. Should be named as Go is naming platforms
+# All supported architectures must have an "ifeq" block below that customizes the parameters
+ALL_ARCHITECTURES=amd64 arm arm64
+ML_PLATFORMS=linux/amd64,linux/arm,linux/arm64
+
+ifeq ($(ARCH),amd64)
+
+	# The architecture and version for use when downloading the docker binary
+	# Lastly the path in the tarball where the docker client binary is placed
+	WEAVEEXEC_DOCKER_ARCH?=x86_64
+	WEAVEEXEC_DOCKER_VERSION=1.6.2
+	WEAVEEXEC_DOCKER_TARBALL_PATH=usr/local/bin/docker
+
+	# The name of the alpine baseimage to use as the base for weave images
+	ALPINE_BASEIMAGE?=alpine
+
+	# The extension for the made images
+	# Specifying none means for example weaveworks/weave:latest
+	ARCH_EXT?=
+
+	# The name of the gcc we're using for compiling C code
+	CC=gcc
+	
+	# Optional parameters that can be passed when linking C code into the Go binary
+	CGO_LDFLAGS=
+endif
+ifeq ($(ARCH),arm)
+	# Here we're using a much newer version of the Docker client
+	# because Docker on ARM hasn't been released in many older versions
+	WEAVEEXEC_DOCKER_ARCH?=armel
+	WEAVEEXEC_DOCKER_VERSION=1.12.3
```

Comment:
Do we really have to pick such a new version?  This will prevent Weave Net being used with 1.11, for instance.

Although #2545 presents an alternative solution.